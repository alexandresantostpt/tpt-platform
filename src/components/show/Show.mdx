---
name: Show
route: /show
---
import { Playground, PropsTable } from 'docz'

import Show from './Show'

import '@cfg/scripts'
import '@cfg/style'

# Show

<PropsTable of={ Show }/>

## Basic usage

```
<Show only={ BOOLEAN_VALUE }>
    <h1>Must be show</h1>
</Show>
```

<Playground>
    <Show only={ true }>
        <h1>Must be show</h1>
    </Show>
</Playground>
