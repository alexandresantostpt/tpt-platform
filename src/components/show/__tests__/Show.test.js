import React from 'react'
import renderer from 'react-test-renderer'

import Show from '@components/show/Show'

describe('Tests for Show component', () => {
    it('Should rendering without crash ', () => {
        const tree = renderer
            .create(
                <Show only={true}>
                    <h1>Must be show</h1>
                </Show>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('Should return null when only is a false value ', () => {
        const tree = renderer
            .create(
                <Show only={false}>
                    <h1>Must not be show</h1>
                </Show>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
