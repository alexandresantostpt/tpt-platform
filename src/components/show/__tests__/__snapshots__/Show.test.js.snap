// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Show component Should rendering without crash  1`] = `
<h1>
  Must be show
</h1>
`;

exports[`Tests for Show component Should return null when only is a false value  1`] = `null`;
